### PR TITLE
Music: sound loading progress bar

### PR DESCRIPTION
**Pull request by breville**
This adds a subtle progress bar for the sound loading in Music Lab.  This doesn't need be a permanent feature, but while we are still loading sounds in the background after the main lab is up and running, it should help give us an intuitive idea about how long this is taking in different situations.  (We are analyzing reported metrics for this, too.)

https://github.com/code-dot-org/code-dot-org/assets/2205926/985d6fad-19f6-4864-b70f-6f0e1c830b85

Coded on the Amtrak to Portland!

### CODE REVIEW

File: apps/src/music/redux/musicRedux.ts [hunk 1]
```diff
@@ -45,6 +41,8 @@ export interface MusicState {
   playbackEvents: PlaybackEvent[];
   /** The current last measure of the song */
   lastMeasure: number;
+  /** The current sound loading progress */
```
Comment (sanchitmalhotra126) [inline apps/src/music/redux/musicRedux.ts]: nit: maybe worth mentioning this number is a fraction of the sounds loaded / total sounds to load?

**Review by sanchitmalhotra126 — COMMENTED** [review]
Neat! This is definitely a helpful feature for debugging. I'd be curious to get @moneppo and @markabarrett's takes on how this would be best exposed to users (i.e. should we just block the whole UI until all sounds have loaded since some may not work?). Would it be worth putting this behind a feature flag for now?

File: apps/src/music/redux/musicRedux.ts (hunk 1 above)
Comment (breville) [reply apps/src/music/redux/musicRedux.ts]: Done.

**Comment by moneppo** [comment]
I think this is good to merge, though I'd like to think through how we could (when we have time) update the UI to make it clear what's happening. I'm currently thinking this should eventually be a blocking loading screen.

**Pull request merged**